mlflow/xgboost.py

```diff
@@ -434,26 +434,75 @@ def log_feature_importance_plot(features, importance, importance_type):
             Log feature importance plot.
             """
             import matplotlib.pyplot as plt
+            from cycler import cycler
 
             features = np.array(features)
-            importance = np.array(importance)
-            indices = np.argsort(importance)
-            features = features[indices]
-            importance = importance[indices]
+
+            # Structure the supplied `importance` values as a `num_features`-by-`num_classes` matrix
+            importances_per_class_by_feature = np.array(importance)
+            if importances_per_class_by_feature.ndim <= 1:
+                # In this case, the supplied `importance` values are not given per class. Rather,
+                # one importance value is given per feature. For consistency with the assumed
+                # `num_features`-by-`num_classes` matrix structure, we coerce the importance
+                # values to a `num_features`-by-1 matrix
+                indices = np.argsort(importance)
+                features = features[indices]
+                importances_per_class_by_feature = np.array(
+                    [[importance] for importance in importances_per_class_by_feature]
+                )
+                # In this case, do not include class labels on the feature importance plot because
+                # only one importance value has been provided per feature, rather than an
+                # one importance value for each class per feature
+                label_classes_on_plot = False
+            else:
+                label_classes_on_plot = True
```

Comment:
Can we sort a 2D importance matrix (that linear boosters generates) as well?

```python
import numpy as np

features = np.array(["a", "b", "c"])
importance = [
    # class0, class1, class2
    [7, 8, 9],  # a
    [4, 5, 6],  # b
    [1, 2, 3],  # c
]
importances_per_class_by_feature = np.array(importance)
abs_sum = np.abs(importances_per_class_by_feature).sum(axis=1)
# or abs_mean = np.abs(importances_per_class_by_feature).mean(axis=1)
indices = np.argsort(abs_sum)

print(importances_per_class_by_feature[indices])
# [[1 2 3]
#  [4 5 6]
#  [7 8 9]]

print(features[indices])
# ['c' 'b' 'a']
```

Reply:
Absolutely! Done! (Chose `sum()` for the magnitude metric rather than `mean()`). Here's a screenshot from a linear booster on MNIST:

![feature_importance_weight](https://user-images.githubusercontent.com/39497902/124563032-8949b580-ddf4-11eb-8ead-0301a3abc295.png)